src/Sulu/Bundle/PageBundle/EventListener/DocumentManagerEventSubscriber.php

```diff
@@ -0,0 +1,526 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\PageBundle\EventListener;
+
+use PHPCR\NodeInterface;
+use Sulu\Bundle\DocumentManagerBundle\Collector\DocumentDomainEventCollectorInterface;
+use Sulu\Bundle\PageBundle\Document\BasePageDocument;
+use Sulu\Bundle\PageBundle\Document\PageDocument;
+use Sulu\Bundle\PageBundle\Domain\Event\PageChildrenReorderedEvent;
+use Sulu\Bundle\PageBundle\Domain\Event\PageCopiedEvent;
+use Sulu\Bundle\PageBundle\Domain\Event\PageCreatedEvent;
+use Sulu\Bundle\PageBundle\Domain\Event\PageDraftRemovedEvent;
+use Sulu\Bundle\PageBundle\Domain\Event\PageModifiedEvent;
+use Sulu\Bundle\PageBundle\Domain\Event\PageMovedEvent;
+use Sulu\Bundle\PageBundle\Domain\Event\PagePublishedEvent;
+use Sulu\Bundle\PageBundle\Domain\Event\PageRemovedEvent;
+use Sulu\Bundle\PageBundle\Domain\Event\PageTranslationAddedEvent;
+use Sulu\Bundle\PageBundle\Domain\Event\PageTranslationCopiedEvent;
+use Sulu\Bundle\PageBundle\Domain\Event\PageTranslationRemovedEvent;
+use Sulu\Bundle\PageBundle\Domain\Event\PageUnpublishedEvent;
+use Sulu\Bundle\PageBundle\Domain\Event\PageVersionRestoredEvent;
+use Sulu\Component\Content\Document\Extension\ExtensionContainer;
+use Sulu\Component\Content\Document\Subscriber\SecuritySubscriber;
+use Sulu\Component\Content\Document\Subscriber\StructureSubscriber;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+use Sulu\Component\DocumentManager\Event\ConfigureOptionsEvent;
+use Sulu\Component\DocumentManager\Event\CopyEvent;
+use Sulu\Component\DocumentManager\Event\CopyLocaleEvent;
+use Sulu\Component\DocumentManager\Event\FlushEvent;
+use Sulu\Component\DocumentManager\Event\MoveEvent;
+use Sulu\Component\DocumentManager\Event\PersistEvent;
+use Sulu\Component\DocumentManager\Event\PublishEvent;
+use Sulu\Component\DocumentManager\Event\RemoveDraftEvent;
+use Sulu\Component\DocumentManager\Event\RemoveEvent;
+use Sulu\Component\DocumentManager\Event\RemoveLocaleEvent;
+use Sulu\Component\DocumentManager\Event\ReorderEvent;
+use Sulu\Component\DocumentManager\Event\RestoreEvent;
+use Sulu\Component\DocumentManager\Event\UnpublishEvent;
+use Sulu\Component\DocumentManager\Events;
+use Sulu\Component\DocumentManager\PropertyEncoder;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Webmozart\Assert\Assert;
+
+class DocumentManagerEventSubscriber implements EventSubscriberInterface
+{
```

Comment:
sorry, did not think about this before. i think we should include `DomainEvent` in the class name because the `DocumentManager` itself is already event base 🙈  maybe just `DomainEventSubscriber`?